api/hub/src/api.ts

```diff
@@ -27,26 +64,295 @@ api.post('/validate-token/:token', async (ctx: koa.Context, next: () => Promise<
+      pubKey: profile.pubKey,
+      data: report.data,
+      signature: report.signature,
+    });
+    if (!verified) {
+      ctx.status = 403;
+    }
+    try {
+      await dataSources.decisionsAPI.makeDecision(
+        report.contentId,
+        report.data.moderator,
+        report.data.explanation,
+        report.data.delisted,
+      );
       ctx.status = 200;
+    } catch (error) {
+      ctx.body = `Cannot moderate content! Error: ${error}`;
+      ctx.status = 500;
+    }
+  }
+  await next();
+});
+
+api.get('/moderation/decisions/:contentId', async (ctx: koa.Context, next: () => Promise<any>) => {
+  const contentID = ctx?.params?.contentId;
+  if (!contentID) {
+    ctx.status = 400;
+    ctx.body = 'Missing "contentId" attribute from request.';
+  } else {
+    ctx.set('Content-Type', 'application/json');
+    ctx.body = await getFinalDecision(contentID);
+    ctx.status = 200;
+  }
+  await next();
+});
+
+api.post('/moderation/decisions/pending', async (ctx: koa.Context, next: () => Promise<any>) => {
+  const req = ctx?.request.body;
+  const decisions = await dataSources.decisionsAPI.listDecisions(
+    false,
+    false,
+    req.offset,
+    req.limit,
+  );
+  const list = await Promise.all(
+    decisions.map(decision => {
+      return getFinalDecision(decision.contentID);
+    }),
+  );
+  ctx.set('Content-Type', 'application/json');
+  ctx.body = list;
+  ctx.status = 200;
+  await next();
+});
+
+api.post('/moderation/decisions/moderated', async (ctx: koa.Context, next: () => Promise<any>) => {
+  const req = ctx?.request.body;
+  if (req.delisted === undefined) {
+    ctx.status = 400;
+    ctx.body = 'Missing "delisted" attribute from request.';
+  } else {
+    const decisions = await dataSources.decisionsAPI.listDecisions(
+      req.delisted,
+      true,
+      req.offset,
+      req.limit,
+    );
+    const list = await Promise.all(
+      decisions.map(decision => {
+        return getFinalDecision(decision.contentID);
+      }),
+    );
+    ctx.set('Content-Type', 'application/json');
+    ctx.body = list;
+    ctx.status = 200;
+  }
+  await next();
+});
+
+api.post('/moderation/moderators/new', async (ctx: koa.Context, next: () => Promise<any>) => {
+  const request = ctx?.request.body;
+  if (!request.data || !request.secret) {
+    ctx.status = 400;
+  } else {
+    let ok = false;
+    if (request.secret === adminSecret) {
+      ok = true;
     } else {
-      ctx.status = 400;
+      const profile = await dataSources.profileAPI.getProfile(request.data.adminEthAddress);
+      if (!profile || !profile.pubKey || !request.data.signature) {
+        ctx.status = 401;
+      } else {
+        try {
+          const verified = await verifyEd25519Sig({
+            pubKey: profile.pubKey,
+            data: request.data,
+            signature: request.signature,
+          });
+          const isAdmin = dataSources.moderatorsAPI.isAdmin(request.data.adminEthAddress);
```

Comment:
I think here you should have `await` or it will be always `true` because the result is a Promise

Reply:
Good catch!